### ISSUE TITLE
Check fails even if all issues got auto fixed

### ISSUE THREAD

**Issue by jonas-arkulpa** [arturbosch/detekt#1836]
## Expected Behavior
If all found issues where automatically fixed the check/build should not fail.

## Observed Behavior
The check/build fails even if after auto fixing no issues exists.
If I run detekt a second time it passes.

## Steps to Reproduce
```
build:
  maxIssues: 1
formatting:
  active: true
  android: true
  autoCorrect: true
```
Introduce an issue which is auto correctable.
Run the detekt gradle task.

## Context
I want to use the same config/task to auto format/fix and check the result
(locally and on the ci server)

## Your Environment
* Version of detekt used: 1.0.0 (also tested with 1.0.0-RC15)
* Version of Gradle used (if applicable): Gradle 5.1.1
* Operating System and version: macOS Mojave

**Comment by schalkms** [comment]
Thanks for reporting this!
I think we shouldn't report issues, which are corrected by the same run of detekt.
However, if the `autoCorrect` is disabled, this issues should be reported.
Greetings to the Ländle 🙂

**Comment by jonas-arkulpa** [comment]
@schalkms thx for the fast help/PR.

Now greetings from the capital. 🙂

**Comment by arturbosch** [comment]
@schalkms I'm not sure but don't we have already the requested behavior by giving the formatting rule set a weight of 0?

```yaml
build:
  maxIssues: 1
  weights:
    formatting: 0
```

**Comment by schalkms** [comment]
I think so, but it's not coupled to the `auto-correct` option.

**Comment by arturbosch** [comment]
We could add an option to the `build`-namespace like `excludeCorrectable` and introduce a new Api `DetektionFilter` or `ResultFilter` which basically should do what FilteredDetectionResult is doing for the baseline feature.

https://github.com/arturbosch/detekt/blob/3543c2fe1aa46b579aac61371a13b9faf23c6e95/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/FilteredDetectionResult.kt#L8-L19

**Comment by schalkms** [comment]
I can't think of a case where you want to add the already fixed issues to the total **maxIssues** count. That's why I don't think this config option is worth the effort.

**Comment by arturbosch** [comment]
> I can't think of a case where you want to add the already fixed issues to the total **maxIssues** count. That's why I don't think this config option is worth the effort.

True thats why we could filter them in a `ResultFilter` is is active by default and print the `ModificationNotification`.

The linked issue has also an use case for `ResultFilter`'s.
